src/components/TodoList/TodoList.tsx

```diff
@@ -0,0 +1,75 @@
+import React, { useContext } from 'react';
+import {
+  TransitionGroup,
+  CSSTransition,
+} from 'react-transition-group';
+import { TodoItem } from '../TodoItem';
+import { StatusEnum } from '../../types/StatusEnum';
+import { Todo } from '../../types/Todo';
+import { FilterTodosContext } from '../../context/TodosContexts';
+
+type Props = {
+  todos: Todo[];
+  tempTodo: Todo | null;
+  onDeleteTodo: (todoId: number) => Promise<void>;
+  onChangeTitle: (todoId: number, newTitle: string) => Promise<void>;
+  onChangeCompletedStatus: (
+    todoId: number,
+    isCompleted: boolean,
+  ) => Promise<void>;
+};
+
+export const TodoList: React.FC<Props> = ({
+  todos,
+  onChangeCompletedStatus,
+  onChangeTitle,
+  onDeleteTodo,
+  tempTodo,
+}) => {
+  const { filter } = useContext(FilterTodosContext);
+
+  return (
+    <section data-cy="TodoList" className="todoapp__main">
+      <TransitionGroup>
+        {todos.filter(todo => {
+          switch (filter) {
+            case StatusEnum.Active:
+              return !todo.completed;
+            case StatusEnum.Completed:
+              return todo.completed;
+            case StatusEnum.All:
+            default:
+              return todo;
+          }
```

Comment:
It is worth putting it in a separate variable and using the finished result to improve the readability of the code.